SWIPs/swip-20-honey_to_money.md

```diff
@@ -0,0 +1,103 @@
+- Zero marginal costs of serving data with perfect competition means that when nodes would engage in peer-to-peer price discovery, the price will tend to zero and a sub-optimal equilibrium of near-zero prices will be reached.
+- The high-quantity of transactions implies that nodes cannot engage in price-discovery for every byte of data transacted, meaning they have to agree *beforehand* on how they will come to a price *at a later time*.
+- Nodes are most of the time just conduits for data and payments, meaning that most of the time data requests (and thus also payments) will take place because nodes want to send data to a peer who requested this data from them.
+
+### Discussion on time differences causing imbalances
+#### Delay between sending and receiving a payment
+Nodes might apply different honey prices for the same payment, due to a time difference between applying a price for *sending* a payment and applying a price for *receiving* the same payment. The imbalances caused by this don't do significant harm if the sum of all imbalances is not greater than the disconnect threshold. Theoretically, if the overall price movement does not show a clear trend (either up or down) over the period where nodes do business with each other, or the two nodes are neither a net receiver nor a net provider to each other, accounting imbalances should offset each other. Furthermore, less frequent as well as small price updates should minimize the imbalances. If a clear trend exists and one node is a net receiver and the other a net provider, the disconnect threshold will eventually be reached. Research must be performed on how the strength of a trend influences the duration until a disconnect is reached for a given stream of traffic and whether this is expected to cause harm to the connectivity in Swarm. The assumption is that this will not be the case for a typical Swarm node. 
+
+If there appears to be a problem with such imbalances, we see a possible solution: as nobody benefits from price imbalances, both the sender and the receiver of a payment are incentivized to apply a price which is not per their clock, but per the clock of the counterparty when they expect the counterparty to query or have queried the oracle. To assist in this, a payload might be sent with the transaction, specifying the clock with which the oracle price was applied.
+
+#### Temporary hard forks
+An on-chain oracle *might* give a different answer to two nodes when queried at the same time if there exists a (temporary) fork in the blockchain. If a disbalance due to such forks becomes common and causes the disconnect threshold to be reached for a significant number of nodes, we may put up additional requirements. For example, a minimum of 12 blocks for confirmations.
```

Comment:
disbalance -> imbalance

Reply:
> For example, a minimum of 12 blocks for confirmations.

Or just use price at `B` blocks ago by default. Or was that what you meant here?

Reply:
> disbalance -> imbalance

Changed.

> Or just use price at B blocks ago by default. Or was that what you meant here?

This is what is meant here, but I will rewrite it to something which is more in line with how you phrased it